### PR TITLE
Remove always-on manage/* feature flags

### PR DESCRIPTION
**Pull request by jsnajdr**
This PR removes several `manage/*` feature flags that are either always-on in all environments or that are not used in code. Most of them have been dormant for years.

**How to test/review:**
Best reviewed commit by commit.

### CODE REVIEW

**Review by sirbrillig — APPROVED** [review]
No objections for the Payments flag. 👍

File: client/my-sites/sidebar/site-menu.jsx [hunk 1]
```diff
@@ -90,7 +90,6 @@ class SiteMenu extends PureComponent {
 				name: 'post',
 				label: translate( 'Posts' ),
 				capability: 'edit_posts',
-				config: 'manage/posts',
```
Comment (tyxla) [inline client/my-sites/sidebar/site-menu.jsx:93]: We should be able to remove the config logic from https://github.com/Automattic/wp-calypso/pull/53455/files#diff-29e15167224ca54ba114cb703ae0632c1101679db5374851b00a0f2bc025cbb5R153 now, right?

File: client/my-sites/sidebar/tools-menu.jsx [hunk 2]
```diff
@@ -49,7 +49,6 @@ class ToolsMenu extends PureComponent {
 			label: translate( 'Plugins' ),
 			capability: 'manage_options',
 			queryable: ! config.isEnabled( 'calypsoify/plugins' ) || ! isAtomicSite,
-			config: 'manage/plugins',
```
Comment (tyxla) [inline client/my-sites/sidebar/tools-menu.jsx:52]: We should be able to remove the config logic on https://github.com/Automattic/wp-calypso/pull/53455/files#diff-0e72ae8e54cb18e6af045f21c5f2f05101cb619e76bc2a39cbcf4ca6f5e4143fR110 right?

**Review by tyxla — APPROVED** [review]
LGTM 👍  Nice cleanup!

File: client/my-sites/sidebar/site-menu.jsx (hunk 1 above)
Comment (jsnajdr) [reply client/my-sites/sidebar/site-menu.jsx:93]: Yes we can! But I need to verify that more carefully and wanted to focus this PR strictly on removing flags that are obviously always true.

Comment (tyxla) [reply client/my-sites/sidebar/site-menu.jsx:93]: Follow-up sounds good to me obviously 👍

**Pull request merged**